src/device_manager.rs

```diff
@@ -3,12 +3,116 @@
 
 //! System level device management.
 //!
-//! [IoManager](struct.IoManager.html) is respondsible for managing
+//! [`IoManager`] is respondsible for managing
 //! all devices of virtual machine, registering IO resources callback,
 //! deregistering devices and helping VM IO exit handling.
-//！VMM would be responsible for getting device resource request, ask
-//! vm_allocator to allocate the resources, ask vm_device to register the
-//! devices IO ranges, and finally set resources to virtual device.
+//! It defines two buses, one for PIO and one for MMIO, and provides default
+//! implementations of [`PioManager`] and [`MmioManager`].
+//!
+//! VMM would be responsible for getting device resource request, allocating the
+//! resources, asking `vm_device` to register the devices IO ranges, and finally
+//! setting resources to virtual device.
+//!
+//! # Examples
+//!
+//! Registering a new device can be done using the register methods of [`PioManager`]
+//! and [`MmioManager`] with an appropriate bus range
+//! ([`PioRange`](../bus/type.PioRange.html) or [`MmioRange`](../bus/type.MmioRange.html)).
+//! ```
+//! # use std::sync::Arc;
+//! # use vm_device::bus::{PioAddress, PioAddressOffset, PioRange};
+//! # use vm_device::bus::{MmioAddress, MmioAddressOffset, MmioRange};
+//! # use vm_device::device_manager::{IoManager, PioManager, MmioManager};
+//! # use vm_device::{DevicePio, DeviceMmio};
```

Comment:
nit: there are lots of imports compared to the number of lines of example code, maybe we can hide them from the example by using ` # ` in front of `use ...`.